lib/modules/manager/git-submodules/extract.ts

```diff
@@ -37,11 +38,16 @@ async function getUrl(
 const headRefRe = regEx(/ref: refs\/heads\/(?<branch>\w+)\s/);
 
 async function getDefaultBranch(subModuleUrl: string): Promise<string> {
-  const val = await Git(simpleGitConfig()).listRemote([
-    '--symref',
-    subModuleUrl,
-    'HEAD',
-  ]);
+  const gitSubmoduleAuthEnvironmentVariables = getGitEnvironmentVariables();
+  const gitEnv = {
+    // pass all existing env variables
+    ...process.env,
```

Comment:
@rarkins Should we use `getChildProcessEnv` here instead?

Reply:
@Shegox pointed out that we already (implicitly) pass the entire env to `git` anyway. So I was thinking it's ok and risky to change?

Reply:
ok

Reply:
we probably should limit it in future, because of possible git hooks